### ISSUE TITLE
Rewriting container config breaks `apt-get update` on Ubuntu 22.04+

### ISSUE THREAD

**Issue by dmartin** [picoCTF/oci-interceptor#69]
For example:

```
root@host:/# docker run --runtime oci-interceptor --rm -it ubuntu:22.04 bash
root@28a94c545cba:/# apt-get update
[...]
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
root@28a94c545cba:/#
```

Something in the post-invoke cleanup scripts must be attempting to write to one of the files we're mounting as read-only (`/etc/hosts`, `/etc/hostname`, `/etc/resolv.conf`)

**Comment by dmartin** [comment]
This actually has nothing to do with the networking mounts being readonly or not. It can be replicated simply by parsing the OCI spec and writing it back to the same path without any modifications

**Comment by dmartin** [comment]
Looking at the `config.json` diff before and after parsing/rewriting with [oci-spec](https://docs.rs/oci-spec/latest/oci_spec/):
Oho, what's this? Our old friend `clone3` that pops up in all the search results regarding this apt-get issue???

<img width="1288" alt="Screenshot 2023-02-27 at 1 01 53 AM" src="https://user-images.githubusercontent.com/1657652/221486953-905e35ba-2636-4087-9c9a-ca16e373250d.png">

**Comment by dmartin** [comment]
will be fixed with https://github.com/containers/oci-spec-rs/pull/125

**Comment by dmartin** [comment]
Fixed in v0.2.0 by depending on a oci-spec-rs fork (https://github.com/dmartin/oci-spec-rs/tree/linuxsyscall-rename).
However, leaving this issue open until we can (hopefully) revert to upstream crate following PR merge.

**Comment by dmartin** [comment]
Resolved w/ https://github.com/picoCTF/oci-interceptor/commit/02abff386e3ca4efc8f45b72821403b435b5544c
